src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -810,6 +810,158 @@ PyObject *Message_new0 (const Handle *handle, const rd_kafka_message_t *rkm) {
 
 
 
+/****************************************************************************
+ *
+ *
+ * Uuid
+ *
+ *
+ *
+ *
+ ****************************************************************************/
+static PyObject *Uuid_most_significant_bits (Uuid *self, PyObject *ignore) {
+        if(self->cUuid) {
+                return cfl_PyLong_FromLong(rd_kafka_Uuid_most_significant_bits(self->cUuid));
+        }
+        return Py_None;
+}
+
+static PyObject *Uuid_least_significant_bits (Uuid *self, PyObject *ignore) {
+        if(self->cUuid) {
+                return cfl_PyLong_FromLong(rd_kafka_Uuid_least_significant_bits(self->cUuid));
+        }
+        return Py_None;
+}
+
+static PyMethodDef Uuid_methods[] = {
+	{ "get_most_significant_bits", (PyCFunction)Uuid_most_significant_bits, METH_NOARGS,
+	  "  :returns: Most significant 64 bits of the 128 bits Uuid"
+	  "  :rtype: int\n"
+	  "\n"
+	},
+	{ "get_least_significant_bits", (PyCFunction)Uuid_least_significant_bits, METH_NOARGS,
+	  "  :returns: Least significant 64 bits of the 128 bits Uuid"
+	  "  :rtype: int\n"
+	  "\n"
+	},
+	{ NULL }
+};
+
+
+static PyObject *Uuid_str0 (Uuid *self) {
+        if(self->cUuid) {
+                const char *base64str = rd_kafka_Uuid_base64str(self->cUuid);
+                if(base64str)
+                        return cfl_PyUnistr(_FromString(base64str));
+        }
+        return Py_None;
+}
+
+static long Uuid_hash (Uuid *self) {
+        
+        return rd_kafka_Uuid_most_significant_bits(self->cUuid) ^ rd_kafka_Uuid_least_significant_bits(self->cUuid);
+}
+
+
+static PyObject *Uuid_new (PyTypeObject *type, PyObject *args,
+                               PyObject *kwargs) {
+        PyObject *self = type->tp_alloc(type, 1);
+        return self;
+}
+
+
+static int Uuid_init (PyObject *self0, PyObject *args,
+                          PyObject *kwargs) {
+        Uuid *self = (Uuid *)self0;
+        static char *kws[] = { "most_significant_bits",
+                               "least_significant_bits",
+                               NULL };
+        int64_t most_significant_bits;
+        int64_t least_significant_bits;
+
+        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "LL", kws,
+                                         &most_significant_bits, 
+                                         &least_significant_bits))
+                return -1;
+
+        self->cUuid = rd_kafka_Uuid_new(most_significant_bits, least_significant_bits);
+
+        return 0;
+}
+
+static int Uuid_clear (Uuid *self) {
+	if (self->cUuid) {
+		rd_kafka_Uuid_destroy(self->cUuid);
+		self->cUuid = NULL;
+	}
+	return 0;
+}
+
+static int Uuid_traverse (Uuid *self,
+                        visitproc visit, void *arg) {
+        return 0;
+}
+
+static void Uuid_dealloc (Uuid *self) {
+        Uuid_clear(self);
+        PyObject_GC_UnTrack(self);
```

Comment:
Should go before Uuid_clear because the documentation says
```
If the type supports garbage collection (has the Py_TPFLAGS_HAVE_GC flag bit set), the destructor should call PyObject_GC_UnTrack() before clearing any member fields.
```